apis/v1alpha1/httproute_types.go

```diff
@@ -267,15 +264,18 @@ type HTTPHeaderMatch struct {
 	// Please read the implementation's documentation to determine the supported
 	// dialect.
 	//
+	// HTTP Header names MUST be always matched case-insensitively (RFC 2616 -
```

Comment:
"case-insensitively" sounds a bit off to me. Maybe something like this would work: "HTTP Header name matching MUST be case-insensitive"